### ISSUE TITLE
takeTill acting wierd

### ISSUE THREAD

**Issue by banacorn** [bos/attoparsec#80]
```haskell
parser :: Parser Text
parser = takeTill ((==) 'a'))

main :: IO ()
main = parseTest parser "𝟘a" >>= print
```

The code should result in `Done "a" "\120792"`, a clean cut.
But I get `Done "\57304a" "\120792"`

With the predicate negated, `takeWhile` also presents the same issue.

The issue can be reproduced with this [gist](https://gist.github.com/banacorn/04aa36742bc651ac2570)
I'm using `attoparsec-0.12.1.2` with `text-1.2.0.0`

Thanks!

**Comment by hesselink** [comment]
Any chance of a release with this fix? I just ran into this with an even simpler reproduction: `takeText "💋"`.

**Comment by banacorn** [comment]
ha, is that a pair of lips?

**Comment by bos** [comment]
Released as 0.12.1.3.

**Comment by hesselink** [comment]
Thanks!
